daffodil-core/src/main/scala/org/apache/daffodil/core/dsom/SchemaDocument.scala

```diff
@@ -212,6 +212,8 @@ final class SchemaDocument private (xmlSDoc: XMLSchemaDocument)
   final override lazy val optLexicalParent = Some(xmlSDoc)
   final override lazy val optXMLSchemaDocument = Some(xmlSDoc)
 
+  final lazy val version = (xml \ "@version").text
```

Comment:
I didn't know XSD had a version attribute. Coudld be useful!

Though, I wonder if this should be an Option so we can tell if it's provided or not? In the code gen case, it can just to `optVersion.getOrElse("")` so it wouldn't change much.

Reply:
Somewhere on WWW is a diatribe on why using namespace URIs with versions in them is so wrong wrong wrong as a versioning means for XSD schemas.

Reply:
How do you change the line to return an Option[String]?

Reply:
> Somewhere on WWW is a diatribe on why using namespace URIs with versions in them is so wrong wrong wrong as a versioning means for XSD schemas.

The built-in XML Schema version attribute is orthogonal to and separate from any namespace URI.  We plan to take advantage of it to assign semantic versions (strings like "3.6.0") to our filters.

Reply:
> How do you change the line to return an Option[String]?

`(xml \ "@version")` returns a `NodeSeq`, which has an implicit conversion to `Seq[Node]`, so you can use any `Seq` functions on it. So something like this should work:

```scala
final lazy val optVersion = (xml \ "@version").headOption.map(_.text)
```

Reply:
Alas, I merged the PR before I checked my emails and saw your reply,  It's too late to put the change into the PR now, but let's keep it in mind for another PR:

```scala
final lazy val optVersion = (xml \ "@version").headOption.map(_.text)
```